x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/pki/PkiRealm.java

```diff
@@ -317,6 +317,17 @@ public void expireAll() {
         }
     }
 
+    @Override
+    public void usageStats(ActionListener<Map<String, Object>> listener) {
+        super.usageStats(ActionListener.wrap(stats -> {
+            stats.put("has_truststore", trustManager != null);
+            stats.put("has_delegated_realms", delegatedRealms != null);
+            stats.put("principal_pattern", principalPattern.pattern());
```

Comment:
I think we should just capture whether this uses the default pattern or not. Leaking that level of site-config info into telemetry seems like it's going too far.

Reply:
Understood, I've made it a boolean field. Thank you :)